src/main/java/ui/components/NavigableListView.java

```diff
@@ -111,25 +114,43 @@ public void restoreSelection() {
                 selectedIndex = Optional.of(nextIndex);
 
                 // The next item will be considered selected
-                onItemSelected.accept(nextIndex);
+                triggerItemSelected(nextIndex);
             }
         }
     }
 
+    /**
+     * Passes the relevant index to the onItemSelected callback variable as well as "false" value as the second
+     * parameter indicating that this method is called for triggering callbacks for left clicks, key presses etc.
+     */
```

Comment:
This description is kind of operational -- we can more or less see that from the code. Maybe you could describe how this works at a higher level of abstraction?